cdap-runtime-ext-dataproc/src/main/java/io/cdap/cdap/runtime/spi/common/DataprocUtils.java

```diff
@@ -58,6 +61,28 @@ public static void deleteGCSPath(Storage storageClient, String bucket, String pa
     }
   }
 
+  /**
+   * Utilty class to parse the keyvalue string from UI Widget and return back HashMap
+   *
+   * @param configValue
+   * @param delimiter
+   * @param keyValueDelimiter
+   * @return
+   */
+  public static Map<String, String> parseKeyValueConfig(String configValue, String delimiter,
+                                                        String keyValueDelimiter) {
+    Map<String, String> map = new HashMap<>();
+    for (String property : configValue.split(delimiter)) {
+      String[] parts = property.split(keyValueDelimiter, 2);
+      if (parts.length == 2) {
```

Comment:
When length is not two, should throws an IllegalArgumentException with appropriate message tell the caller what's wrong and how to correct it.

Reply:
Done